zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -365,7 +364,25 @@ where
         }?
     }
 
-    let (tx, tx_metadata) = builder.build(&prover).map_err(Error::Builder)?;
+    let fee_strategy = FixedFeeStrategy::new(DEFAULT_FEE);
```

Comment:
Line 429 below always stores `fee_amount: DEFAULT_FEE` in the wallet db regardless of the strategy.